lib/agent0/agent0/interactive_fuzz/helpers/execute_random_trades.py

```diff
@@ -0,0 +1,160 @@
+
+def execute_random_trades(
+    num_trades: int,
+    chain: LocalChain,
+    rng: Generator,
+    interactive_hyperdrive: InteractiveHyperdrive,
+    advance_time: bool = False,
+) -> list[tuple[InteractiveHyperdriveAgent, OpenLong | OpenShort]]:
+    """Conduct some trades specified by the trade list.
+    If advance time is true, the sum of all time passed between all trades will be between 0 and the position duration.
+
+    Arguments
+    ---------
+    num_trades: int
+        The number of trades to execute.
+    chain: LocalChain
+        An instantiated LocalChain.
+    rng: `Generator <https://numpy.org/doc/stable/reference/random/generator.html>`_
+        The numpy Generator provides access to a wide range of distributions, and stores the random state.
+    interactive_hyperdrive: InteractiveHyperdrive
+        An instantiated InteractiveHyperdrive object.
+    advance_time: bool, optional
+        If True, advance time a random amount between 0 and the position duration after each trade.
+        Defaults to False.
+
+    Returns
+    -------
+    list[tuple[InteractiveHyperdriveAgent, OpenLong | OpenShort]]
+        A list with an entry per trade, containing a tuple with:
+            - the agent executing the trade
+            - either the OpenLong or OpenShort trade event
+    """
+    if advance_time:
+        # Generate the total time elapsed for all trades
+        max_advance_time = rng.integers(
+            low=0, high=interactive_hyperdrive.hyperdrive_interface.pool_config.position_duration
+        )
+        # Generate intermediate points between 0 and max_advance_time, sorted in ascending order
+        # Generating number of trades + 1 since cumulative diff results in one less
+        intermediate_points = np.sort(rng.integers(low=0, high=max_advance_time, size=num_trades + 1))
+        # Find cumulative differences of intermediate points for how much time to wait between each trade
+        time_diffs = np.diff(intermediate_points)
+
+    # Generate a list of trades
+    available_actions = np.array([HyperdriveActionType.OPEN_LONG, HyperdriveActionType.OPEN_SHORT])
+    # Do the trades
+    trade_events: list[tuple[InteractiveHyperdriveAgent, OpenLong | OpenShort]] = []
+    for trade_index, trade_type in enumerate([rng.choice(available_actions, size=1)[0] for _ in range(num_trades)]):
+        trade_amount = _get_open_trade_amount(trade_type, rng, interactive_hyperdrive)
+        # extra base accounts for fees
+        # the short trade amount is technically bonds, but we know that will be less than the required base
+        agent = interactive_hyperdrive.init_agent(base=trade_amount * 2, eth=FixedPoint(100))
+        trade_event = _execute_trade(trade_type, trade_amount, agent)
+        trade_events.append((agent, trade_event))
+        if advance_time:
+            # Advance a random amount of time between opening trades
+            chain.advance_time(
+                time_diffs[trade_index],  # type: ignore
```

Comment:
This type ignore is complaining that time_diffs is possibly unbound, because the variable is in an if statement above. Define the variable and run an assertion to avoid this type:ignore